### PR TITLE
opa: new package

### PR DESCRIPTION
**Pull request by Pyronewbic**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: opa.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    description: Opa binary symlink for Neuvector
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/usr/local/bin
+          ln -s /usr/bin/opa ${{targets.contextdir}}/usr/local/bin/opa    
+
+test:
+  pipeline:
+    - runs: |
+        opa --help
```
Comment (EyeCantCU) [inline opa.yaml:51]: Can we do any more here to validate OPA works?

Comment (Pyronewbic) [reply opa.yaml:51]: They're all related to rego test cases, which means we''d have to copy the whole thing over

File: opa.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    strip-prefix: v
+    use-tag: true
+
+subpackages:
+  - name: ${{package.name}}-compat
+    description: Opa binary symlink for Neuvector
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}/usr/local/bin
+          ln -s /usr/bin/opa ${{targets.contextdir}}/usr/local/bin/opa
```
Comment (jamonation) [inline opa.yaml:46]:
```suggestion
```

Let's ditch this and create a symlink over in the main neuvector package(s) that depend on opa.

Comment (Pyronewbic) [reply opa.yaml]: Works for me - it's just not needed compact packages otherwise.

File: opa.yaml (hunk 1 above)
Comment (jamonation) [reply opa.yaml:44]: I think we could include the example.rego and example_test.rego files from https://www.openpolicyagent.org/docs/latest/policy-testing/ as a quick way to validate `opa` can compile and run policies - I'd put them in the `opa` directory so they're available in the test pipeline.

File: opa.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      }
+
+      test_get_user_allowed if {
+        allow with input as {"path": ["users", "bob"], "method": "GET", "user_id": "bob"}
+      }
+
+      test_get_another_user_denied if {
+        not allow with input as {"path": ["users", "bob"], "method": "GET", "user_id": "alice"}
+      }
+      EOF
```
Comment (jamonation) [inline opa.yaml:71]:
```suggestion
```

These two files can go in a directory called `opa` - e.g. `opa/example.rego` and `opa/example_test.rego`.

Then the test pipeline can invoke them directly.

File: opa.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  github:
+    identifier: open-policy-agent/opa
+    strip-prefix: v
+    use-tag: true
+
+test:
+  pipeline:
+    - runs: |
+        opa version
+        cd /opa && opa test . -v
```
Comment (jamonation) [inline opa.yaml:84]:
```suggestion
        opa test . -v
```

If the example files live in the `opa` directory here in the repo, then they will be available in the test pipeline. Here's an example showing them being invoked:

```
2024/05/02 15:52:48 WARN + opa test . -v
2024/05/02 15:52:48 INFO example_test.rego:
2024/05/02 15:52:48 INFO data.authz.test_post_allowed: PASS (374.173µs)
2024/05/02 15:52:48 INFO data.authz.test_get_anonymous_denied: PASS (157.69µs)
2024/05/02 15:52:48 INFO data.authz.test_get_user_allowed: PASS (215.944µs)
2024/05/02 15:52:48 INFO data.authz.test_get_another_user_denied: PASS (166.987µs)
2024/05/02 15:52:48 INFO --------------------------------------------------------------------------------
2024/05/02 15:52:48 INFO PASS: 4/4
```

**Review by jamonation — CHANGES_REQUESTED** [review]

**Review by jamonation — APPROVED** [review]
CVE is a false-positive, so +1 here

**Pull request merged**